system/lib/libcxxabi/src/cxa_exception_emscripten.cpp

```diff
@@ -130,3 +130,21 @@ char* __get_exception_terminate_message(void* thrown_object) {
 }
 
 #endif // __USING_EMSCRIPTEN_EXCEPTIONS__ || __USING_WASM_EXCEPTIONS__
+
+#ifndef __USING_WASM_EXCEPTIONS__
+
+namespace __cxxabiv1 {
+
+void* __cxa_allocate_exception(size_t size) _NOEXCEPT {
+  // Thrown object is prepended by exception metadata block
+  __cxa_exception* ex = (__cxa_exception*)malloc(size + sizeof(__cxa_exception));
```

Comment:
The JS version did `2*` on that sizeof. Was it always allocating too much?

Reply:
No .. the JS version allocated just 1 * sizeof and then incremented the resulting pointer by sizeof.

In the native code we have a helper function that does the increment.

Reply:
Ah, yes... my eyes didn't see the `)` in the sea of `}}}`s :laughing:

Reply:
I wrote the exactly same question and then realized it wasn't what I thought...